app/controllers/benefits/ProfessionalSubscriptionsBenefitsAmountController.scala

```diff
@@ -0,0 +1,107 @@
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+
+class ProfessionalSubscriptionsBenefitsAmountController @Inject()(implicit val cc: MessagesControllerComponents,
+                                                                  authAction: AuthorisedAction,
+                                                                  inYearAction: InYearAction,
+                                                                  appConfig: AppConfig,
+                                                                  professionalSubscriptionBenefitsView: ProfessionalSubscriptionsBenefitsAmountView,
+                                                                  val employmentSessionService: EmploymentSessionService,
+                                                                  errorHandler: ErrorHandler,
+                                                                  ec: ExecutionContext,
+                                                                  clock: Clock) extends FrontendController(cc) with I18nSupport with SessionHelper with FormUtils {
+
+  private def buildForm(isAgent: Boolean): Form[BigDecimal] = {
+    AmountForm.amountForm(s"benefits.professionalSubscriptionsAmount.error.noEntry.${if (isAgent) "agent" else "individual"}",
+      s"benefits.professionalSubscriptionsAmount.error.wrongFormat.${if (isAgent) "agent" else "individual"}",
+      s"benefits.professionalSubscriptionsAmount.error.overMaximum.${if (isAgent) "agent" else "individual"}")
+  }
+
+  private def redirects(cya: EmploymentCYAModel, taxYear: Int, employmentId: String) = {
+    RedirectService.employerProvidedSubscriptionsBenefitsAmountRedirects(cya, taxYear, employmentId)
+  }
+
+  def show(taxYear: Int, employmentId: String): Action[AnyContent] = authAction.async { implicit user =>
+    inYearAction.notInYear(taxYear) {
+      employmentSessionService.getAndHandle(taxYear, employmentId) { (optCya, prior) =>
+        redirectBasedOnCurrentAnswers(taxYear, employmentId, optCya, EmploymentBenefitsType)(redirects(_, taxYear, employmentId)) { cya =>
+
+          val cyaAmount = cya.employment.employmentBenefits.flatMap(_.utilitiesAndServicesModel.flatMap(_.employerProvidedProfessionalSubscriptions))
+          val form = fillFormFromPriorAndCYA(buildForm(user.isAgent), prior, cyaAmount, employmentId) { employment =>
+            employment.employmentBenefits.flatMap(_.benefits.flatMap(_.employerProvidedProfessionalSubscriptions))
+          }
+
+          Future.successful(Ok(professionalSubscriptionBenefitsView(taxYear, form, cyaAmount, employmentId)))
+        }
+      }
+    }
+  }
+
+  def submit(taxYear: Int, employmentId: String): Action[AnyContent] = authAction.async { implicit user =>
+    inYearAction.notInYear(taxYear) {
+
+      employmentSessionService.getSessionDataAndReturnResult(taxYear, employmentId)(CheckYourBenefitsController.show(taxYear, employmentId).url) { cya =>
```

Comment:
I know some pages do use getSessionDataAndReturnResult but could use getSessionDataResult instead so that we could remove the need for the passed in benefits CYA redirect url.